spec/models/user_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validation' do
+    context 'presence' do
+      it { should validate_presence_of(:username) }
+
+      it { should validate_presence_of(:password) }
+
+      it { should validate_presence_of(:password_confirmation) }
+    end
+
+    context 'uniqueness' do
+      subject { User.new(username: 'nimblehq', password: 'password', password_confirmation: 'password') }
```

Comment:
We can use the fabricator intead
```suggestion
      subject { Fabricate(:user, username: 'nimblehq', password: 'password', password_confirmation: 'password') }
```

Reply:
Why do we need to create a new user here, but the others are not?

Reply:
because `validate_uniqueness_of` works by matching a new record against an existing record.

ref : error when try to validate uniqueness without create new user before

<img width="762" alt="Screen Shot 2563-09-24 at 11 58 47" src="https://user-images.githubusercontent.com/29707647/94102855-503fe600-fe5d-11ea-9078-6308f151c00b.png">

Reply:
Fixed to use fabricator on 5955d6440f321194b07b8ccf40d3bbcac5f07cc7